avocado/core/job.py

```diff
@@ -275,6 +275,9 @@ def _make_test_suite(self, references=None):
         loader.loader.load_plugins(self.args)
         try:
             suite = loader.loader.discover(references)
```

Comment:
Hmm, the way I'd imagine this feature was that the loader itself supports the filtering to avoid creating a big list of tests which would be then filtered. I remember you talked about a generic concept of filtering, I thought you'll incorporate it along with the method filtering. Did you change your mind, or did I understood it incorrectly in the first place?